From: Carlos O'Donell <carlos@redhat.com>
Date: Thu, 29 Aug 2013 04:17:33 +0000 (-0400)
Subject: ARM: Pass dl_hwcap to IFUNC resolver.
X-Git-Url: https://sourceware.org/git/?p=glibc.git;a=commitdiff_plain;h=d0721e703d222c01a9e8c329311c4fb01dac6972;hp=3a3092398356c90788479fa75524a491e9d860cd

ARM: Pass dl_hwcap to IFUNC resolver.

For REL relocs pass dl_hwcap to the IFUNC resolver
as is required by the IFUNC API (bug 15905).
---

diff --git a/ports/sysdeps/arm/dl-machine.h b/ports/sysdeps/arm/dl-machine.h
index d251527..85dba67 100644
--- a/ports/sysdeps/arm/dl-machine.h
+++ b/ports/sysdeps/arm/dl-machine.h
@@ -503,7 +503,7 @@ elf_machine_rel (struct link_map *map, const Elf32_Rel *reloc,
 	  break;
 	case R_ARM_IRELATIVE:
 	  value = map->l_addr + *reloc_addr;
-	  value = ((Elf32_Addr (*) (void)) value) ();
+	  value = ((Elf32_Addr (*) (int)) value) (GLRO(dl_hwcap));
 	  *reloc_addr = value;
 	  break;
 #endif

